service/history/historyEngine.go

```diff
@@ -2407,6 +2425,32 @@ func (e *historyEngineImpl) SignalWithStartWorkflowExecution(
 	)
 }
 
+func (e *historyEngineImpl) checkForHistoryCorruptions(ctx context.Context, mutableState execution.MutableState) error {
+	domainName := mutableState.GetDomainEntry().GetInfo().Name
+	if !e.config.EnableHistoryCorruptionCheck(domainName) {
```

Comment:
I know this is demonstration code only, but perhaps a log/metric would help the rollout of this feature. If it's not enabled, at least we could get a rough idea of the number of events.

Reply:
We already have a rough estimates, as we see those errors once events get replicated and put into DLQ.